lib/miq_expression/remove_equality_visitor.rb

```diff
@@ -0,0 +1,71 @@
+class MiqExpression
+  # rubocop:disable Naming/MethodName
+  class RemoveEqualityVisitor < Arel::Visitors::Reduce
+    def visit_NilClass(object, _collector)
+      object
+    end
+
+    def visit_Arel_Nodes_Equality(object, collector)
+      field, binds = collector
+      if (object.left == field.left && object.right == field.right) ||
+         (object.right == field.left && object.left == field.right)
+        # remove this node from the query
+        nil
+      elsif object.right.kind_of?(Arel::Nodes::BindParam)
+        # replace this node with "field = $(next value from the binds array)"
+        object.left.eq(binds.shift.value_for_database)
+      else
+        # leave this node in the query unchanged
+        object
+      end
+    end
+
+    def visit_Arel_Nodes_And(object, collector)
```

Comment:
Do we need to implement this ourselves?  Can't we just inherit from the `ToSql` visitor and avoid having to implement this method?

That aside, I am also not sure what this is doing that is special and needs to be implemented by us.  Is there something that I am missing?

Reply:
Seems like `Arel::Visitors::PostgreSQL` was used in the previous visitor (now deleted)

Reply:
ToSql adds strings to the sql things like "AND" and OR
I suppose Descendant parser would probably be better

I could see how that turns out.

Reply:
truth be told, I'm reluctant to use sql or even descent visitors. Nothing was built to allow someone to manipulate the tree. All of them are basically to produce sql.